editoast/src/models/rolling_stock.rs

```diff
@@ -0,0 +1,144 @@
+            Ok(RollingStockWithLiveries {
+                rolling_stock: self.into(),
+                liveries,
+            })
+        })
+        .await
+        .unwrap()
+    }
+}
+
+impl From<RollingStockModel> for RollingStock {
+    fn from(rolling_stock_model: RollingStockModel) -> Self {
+        RollingStock {
+            id: rolling_stock_model.id.unwrap(),
+            name: rolling_stock_model.name.unwrap(),
+            version: rolling_stock_model.version.unwrap(),
+            effort_curves: rolling_stock_model.effort_curves.unwrap(),
+            base_power_class: rolling_stock_model.base_power_class.unwrap(),
+            length: rolling_stock_model.length.unwrap(),
+            max_speed: rolling_stock_model.max_speed.unwrap(),
+            startup_time: rolling_stock_model.startup_time.unwrap(),
+            startup_acceleration: rolling_stock_model.startup_acceleration.unwrap(),
+            comfort_acceleration: rolling_stock_model.comfort_acceleration.unwrap(),
+            gamma: rolling_stock_model.gamma.unwrap(),
+            inertia_coefficient: rolling_stock_model.inertia_coefficient.unwrap(),
+            features: rolling_stock_model.features.unwrap(),
+            mass: rolling_stock_model.mass.unwrap(),
+            rolling_resistance: rolling_stock_model.rolling_resistance.unwrap(),
+            loading_gauge: rolling_stock_model.loading_gauge.unwrap(),
+            metadata: rolling_stock_model.metadata.unwrap(),
+            power_restrictions: rolling_stock_model.power_restrictions.unwrap(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::client::PostgresConfig;
+    use crate::models::{Create, Delete, Retrieve};
+    use crate::schema::rolling_stock::RollingStock;
+    use actix_web::test as actix_test;
+    use actix_web::web::Data;
+    use diesel::r2d2::{ConnectionManager, Pool};
+    use diesel::PgConnection;
+
+    use super::RollingStockModel;
+
+    #[actix_test]
+    async fn create_get_delete_rolling_stock() {
+        let manager = ConnectionManager::<PgConnection>::new(PostgresConfig::default().url());
+        let db_pool = Data::new(Pool::builder().max_size(1).build(manager).unwrap());
+
+        let rolling_stock: RollingStockModel =
+            serde_json::from_str(include_str!("../tests/example_rolling_stock.json"))
+                .expect("Unable to parse");
+
+        // create a rolling stock
+        let rolling_stock: RollingStock =
+            rolling_stock.create(db_pool.clone()).await.unwrap().into();
+        let rolling_stock_id = rolling_stock.id;
+
+        // get a rolling stock
+        assert!(
+            RollingStockModel::retrieve(db_pool.clone(), rolling_stock_id)
+                .await
+                .unwrap()
+                .is_some()
+        );
+
+        // delete a rolling stock
+        assert!(RollingStockModel::delete(db_pool.clone(), rolling_stock_id)
+            .await
+            .unwrap());
+    }
```

Comment:
You could use transactions (same for all the tests), and maybe split these tests.

Reply:
Done